File: app/src/main/java/com/spiderbiggen/randomchampionselector/model/Ability.java
package com.spiderbiggen.randomchampionselector.model;

/**
 * Created by Stefan on 22-5-2015.
 */
public class Ability {
    private String name;
    private String cost;
    private String cooldown;
    private String description;

    /**
     * Default constructor.
     *
     * @param name        the name of the ability
     * @param cost        the ability cost
     * @param cooldown    the cool down
     * @param description the description
     */
    public Ability(String name, String cost, String cooldown, String description) {
        this.name = name;
        this.cost = cost;
        this.cooldown = cooldown;
        this.description = description;
    }


    public String getName() {
        return name;
    }

    public String getCost() {
        return cost;
    }

    public String getCooldown() {
        return cooldown;
    }

    public String getDescription() {
        return description;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        Ability ability = (Ability) o;

        return getName().equals(ability.getName()) &&
            !(getCost() != null
                ? !getCost().equals(ability.getCost())
                : ability.getCost() != null) &&
            !(getDescription() != null
                ? !getDescription().equals(ability.getDescription())
                : ability.getDescription() != null);

    }

    @Override
    public int hashCode() {
        int result = getName().hashCode();
        result = 31 * result + (getCost() != null ? getCost().hashCode() : 0);
        result = 31 * result + (getDescription() != null ? getDescription().hashCode() : 0);
        return result;
    }
}

File: app/src/main/java/com/spiderbiggen/randomchampionselector/domain/Ability.kt
package com.spiderbiggen.randomchampionselector.domain

/**
 *
 * @author Stefan Breetveld
 *
 * @param name        the name of the ability
 * @param cost        the ability cost
 * @param cooldown    the cool down
 * @param description the description
 */
data class Ability(val name: String, val cost: String?, val cooldown: String, val description: String) {

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || other !is Ability) {
            return false
        }

        val ability = other as? Ability

        return name == ability!!.name &&
                cost == ability.cost &&
                description == ability.description

    }

    override fun hashCode(): Int {
        var result = name.hashCode()
        result = 31 * result + (cost?.hashCode() ?: 0)
        result = 31 * result + (description.hashCode())
        return result
    }
}
